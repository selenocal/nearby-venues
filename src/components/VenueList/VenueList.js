import React from "react";
import VenueItem from "../VenueItem/VenueItem";
import styled from "styled-components";
import PropTypes from "prop-types";

const VenueList = ({ venues }) => (
  <Wrapper>
    {venues.map(venueItem => (
      <VenueItem key={venueItem.venue.id} venue={venueItem.venue} />
    ))}
  </Wrapper>
);

const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

VenueList.propTypes = {
  venues: PropTypes.array
};

VenueList.defaultProps = {
  venues: []
};

export default VenueList;
