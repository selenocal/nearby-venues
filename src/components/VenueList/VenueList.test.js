import React from "react";
import { shallow, mount } from "enzyme";
import VenueList from "./VenueList";

describe("<VenueList />", () => {
  const venueData = [
    {
      venue: {
        id: "1234",
        name: "Adyen Cafe",
        location: {
          address: "Amsterdam"
        },
        categories: []
      }
    },
    {
      venue: {
        id: "5678",
        name: "Adyen Office",
        location: {
          address: "Amsterdam Centraal"
        },
        categories: []
      }
    }
  ];

  it("expects render empty VenueList correctly", () => {
    const wrapper = shallow(<VenueList />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expects render VenueList with mock data correctly", () => {
    const wrapper = mount(<VenueList venues={venueData} />);
    expect(wrapper).toMatchSnapshot();
    venueData.forEach(venueItem => {
      expect(wrapper.contains(venueItem.venue.name)).toBe(true);
      expect(wrapper.contains(venueItem.venue.location.address)).toBe(true);
    });
  });
});
