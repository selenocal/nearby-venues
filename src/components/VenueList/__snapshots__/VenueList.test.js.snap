// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VenueList /> expects render VenueList with mock data correctly 1`] = `ReactWrapper {}`;

exports[`<VenueList /> expects render empty VenueList correctly 1`] = `ShallowWrapper {}`;
